src/index.js

```diff
@@ -32,26 +32,26 @@ function isPrimitive (value) {
 function monadic (fn, cache, serializer, arg) {
   var cacheKey = isPrimitive(arg) ? arg : serializer(arg)
 
-  if (!cache.has(cacheKey)) {
-    var computedValue = fn.call(this, arg)
+  var computedValue = cache.get(cacheKey);
+  if (typeof computedValue==="undefined") {
```

Comment:
This seems to be sacrificing API correctness for speed.

Reply:
@megawac I would say mots of my path does sacrifice API correctness for speed, but the stated goal of the library is to provide the "fastest possible memoization library in JavaScript that supports N arguments." I was just working in concert with its stated goal. Also, the failure to treat string keys correctly was in my opinion a fundamental semantic/functional flaw. Take a look at: https://codeburst.io/a-key-to-keys-when-javascript-keys-dont-match-ab44c81adc87.